src/expanduser.cpp

```diff
@@ -200,4 +198,71 @@ boost::filesystem::path ExpandUser(const std::string& path)
     return {ReplaceString(path, "~", home_dir)};
 }
 
+#else
+
+namespace {
+std::optional<std::string> GetEnvironmentVariable(const std::string_view name)
+{
+    std::size_t required_size;
+    getenv_s(&required_size, nullptr, 0, name.data());
+    if(required_size == 0)
+    {
+        return std::nullopt;
+    }
+    // getenv_s returns the required size of a string including '\0' character.
+    std::string result(required_size - 1, 'A');
+    getenv_s(&required_size, result.data(), required_size, name.data());
+    return {result};
+}
```

Comment:
Probably it duplicates functionality of `<miopen/env.hpp>`.
Can you consider using it, or can you propose how it can be improved to fit windows functionality?

Reply:
@apwojcik could you resolve this question before merging this PR?